nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/io/IsSharedMainContentJsonEnricher.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * (C) Copyright 2024 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Guillaume Renard
+ */
+package org.nuxeo.coldstorage.io;
+
+import static org.nuxeo.coldstorage.ColdStorageConstants.FILE_CONTENT_PROPERTY;
+import static org.nuxeo.ecm.core.io.registry.reflect.Instantiations.SINGLETON;
+import static org.nuxeo.ecm.core.io.registry.reflect.Priorities.REFERENCE;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.io.marshallers.json.enrichers.AbstractJsonEnricher;
+import org.nuxeo.ecm.core.io.registry.context.RenderingContext.SessionWrapper;
+import org.nuxeo.ecm.core.io.registry.reflect.Setup;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+/**
+ * Enricher that tells whether a document has its main content referenced by another document.
+ *
+ * @since 2023.4
+ */
+@Setup(mode = SINGLETON, priority = REFERENCE)
+public class IsSharedMainContentJsonEnricher extends AbstractJsonEnricher<DocumentModel> {
+
+    public static final String NAME = "isSharedMainContent";
+
+    public IsSharedMainContentJsonEnricher() {
+        super(NAME);
+    }
+
+    @Override
+    public void write(JsonGenerator jg, DocumentModel document) throws IOException {
+        jg.writeBooleanField(NAME, isDeduplicated(document));
+    }
+
+    protected boolean isDeduplicated(DocumentModel document) throws IOException {
```

Comment:
Could we keep a consistent naming and use `isSharedContent` instead of `isDeduplicated` which is hard to understand